jsapp/js/components/modalForms/formMedia.es6

```diff
@@ -0,0 +1,280 @@
+import React from 'react';
+import autoBind from 'react-autobind';
+import alertify from 'alertifyjs';
+import Dropzone from 'react-dropzone';
+import TextBox from 'js/components/common/textBox';
+import {ASSET_FILE_TYPES} from '../../constants';
+import {actions} from '../../actions';
+import {bem} from 'js/bem';
+import {LoadingSpinner} from 'js/ui';
+
+import {
+  truncateString,
+  truncateUrl,
+} from '../../utils';
+
+const MAX_ITEM_LENGTH = 50;
+
+/**
+ * @prop {object} asset
+ *
+ * Modal for uploading form media
+ */
+class FormMedia extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      uploadedAssets: [],
+      fieldsErrors: {},
+      inputURL: '',
+      // to show loading icon instead of nothing on first load
+      isInitialised: false,
+      // to show loading icon while uploading any file
+      isUploadFilePending: false,
+      isUploadURLPending: false
+    };
+
+    autoBind(this);
+  }
+
+  /*
+   * setup
+   */
+
+  componentDidMount() {
+    actions.media.loadMedia(this.props.asset.uid);
+
+    actions.media.loadMedia.completed.listen(this.onGetMediaCompleted);
+    actions.media.uploadMedia.failed.listen(this.onUploadFailed);
+  }
+
+  /*
+   * action listeners
+   */
+
+  onGetMediaCompleted(uploadedAssets) {
+    this.setState({
+      uploadedAssets: uploadedAssets.results,
+      isUploadFilePending: false,
+      isUploadURLPending: false,
+      isInitialised: true,
+    });
+  }
+
+  onUploadFailed(response) {
+    this.setState({
+      fieldsErrors: response.responseJSON,
+      isUploadFilePending: false,
+      isUploadURLPending: false
+    });
+  }
+
+  /*
+   * Utilities
+   */
+
+  toBase64(file) {
+    return new Promise(function(resolve, reject) {
+      var reader = new FileReader();
+      reader.readAsDataURL(file);
+      reader.onload = function() {
+          return resolve(reader.result);
+      };
+      reader.onerror = function(error) {
+          return reject(error);
+      };
+    });
+  }
+
+  uploadMedia(formMediaJSON) {
+    // Reset error message before uploading again
+    this.setState({fieldsErrors: {}});
+    actions.media.uploadMedia(this.props.asset.uid, formMediaJSON);
+  }
+
+  /*
+   * DOM listeners
+   */
+
+  onFileDrop(files) {
+    if (files.length >= 1) {
+      this.setState({isUploadFilePending: true});
+
+      files.forEach(async (file) => {
+        var base64File = await this.toBase64(file);
+
+        this.uploadMedia({
+	      description: 'default',
```

Comment:
Finally found why I added this, turns out it can be any string :D
https://github.com/kobotoolbox/kpi/blob/a5302750fa0974d075495bfc070afc85ff5cf60d/kpi/views/v2/asset_file.py#L49-L67